implementation/src/main/java/io/smallrye/mutiny/operators/multi/builders/BufferItemMultiEmitter.java

```diff
@@ -33,7 +33,7 @@ public MultiEmitter<T> emit(T t) {
             fail(new NullPointerException("`emit` called with `null`."));
             return this;
         }
-        if (queue.offer(t) && !Queues.isOverflowing(queue, overflowBufferSize)) {
+        if (queue.offer(t) && (overflowBufferSize == -1 || strictBoundCounter.incrementAndGet() < overflowBufferSize)) {
```

Comment:
Especially for the Spsc scenario, can be quite costy to execute this additional atomic op, which will likely contend on the consumption side too (because the counter is meant to be shared with consumer-s too!).

I suggest instead to have an Overflow Strategy API which enable few nice optimizations, at least for the Single-Producer Single-Consumer case which is super common 
eg using 2 separate counters for submission/consumption both using `lazySet(get() +/- 1)` which is way cheaper than using a contended operation over the same atomic instance.

One of the reasons of using JCTools, together with the nice effect to no longer maintain concurrent tests which prove correctness of the algorithm, is that they have specialized queues based on use cases, but sadly having few costy atomic operations in a row, at least for x86, can partially negate this additional efficiency benefits.

Reply:
The queues here are MPSC ones